rust/main/agents/relayer/src/relayer.rs

```diff
@@ -1408,4 +1508,182 @@ mod test {
             .unwrap();
         assert_eq!(metric.get(), 1);
     }
+
+    async fn build_relayer(settings: RelayerSettings) -> eyre::Result<Relayer> {
+        let agent_metadata = AgentMetadata::new("relayer_git_hash".into());
+
+        let metrics = settings.as_ref().metrics("relayer")?;
+        let task_monitor = tokio_metrics::TaskMonitor::new();
+        let agent_metrics = AgentMetrics::new(&metrics)?;
+        let chain_metrics = ChainMetrics::new(&metrics)?;
+        let runtime_metrics = RuntimeMetrics::new(&metrics, task_monitor)?;
+
+        let (_, tokio_server) = console_subscriber::ConsoleLayer::new();
+
+        Relayer::from_settings(
+            agent_metadata,
+            settings,
+            metrics,
+            agent_metrics,
+            chain_metrics,
+            runtime_metrics,
+            tokio_server,
+        )
+        .await
+    }
+
+    /// Run relayer for 50s to ensure it doesn't crash
+    async fn test_run_relayer(agent: Relayer) -> Result<(), Elapsed> {
+        let future = agent.run();
+        tokio::time::timeout(Duration::from_secs(50), future).await
+    }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_from_settings_and_run_happy_path() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let db_path = temp_dir.path();
+        let chains = vec![(
+            "arbitrum".to_string(),
+            generate_test_chain_conf(
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                None,
+                "https://sepolia-rollup.arbitrum.io/rpc",
+            ),
+        )];
+        let origin_chains = &[HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum)];
+        let destination_chains = &[HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum)];
+        let settings =
+            generate_test_relayer_settings(db_path, chains, origin_chains, destination_chains);
+
+        let agent = build_relayer(settings)
+            .await
+            .expect("Failed to build relayer");
+
+        assert!(test_run_relayer(agent).await.is_err());
```

Comment:
Imo wanting an err is a bit unintuitive, is there a nicer way to do this? ideally where we assert that things are Ok

Reply:
okay :+1: updated